examples/decomposition/plot_kernel_pca_approximate.py

```diff
@@ -0,0 +1,128 @@
+"""
+========================
+Kernel PCA Approximation
+========================
+
+This example shows that using a limited number of components in Kernel PCA can
+lead to a good approximation while being much faster to execute for large
+datasets.
+
+Description:
+------------
+2000 Random data samples are generated similarly to the "Kernel PCA" example
+(the circles). A KernelPCA model is fit on that data with a reduced number of
+principal components, varying from 4 to 1999. For each value, data is
+transformed and inverse-transformed.
+
+Original data is displayed in top left corner. Then subsequent plots in the
+top row show approximations of this data made by KernelPCA transform + inverse
+transform, for various number of components. The bottom row shows the data
+samples projected along the first two principal components for each case.
+
+What you can observe:
+---------------------
+With quite small `n_components`, the approximation is almost identical to the
+full KernelPCA using 2000 components, while execution is much faster
+(execution times are included in the plot titles)
+
+Going further:
+--------------
+You can have a look at the `Kernel PCA Solvers comparison benchmarks
+<https://scikit-learn.org/stable/modules/decomposition.html#kernel-pca-solvers-benchmarks>`_
+to compare execution times in more details.
+"""
+print(__doc__)  # noqa
+
+# Authors: Sylvain MARIE, Schneider Electric
+# License: BSD 3 clause
+
+from datetime import datetime
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.decomposition import KernelPCA
+from sklearn.datasets import make_circles
+
+
+# 1- Generate random data
+# -----------------------
+np.random.seed(0)
+n_samples = 2000
+X, y = make_circles(n_samples=n_samples, factor=.3, noise=.05)
+
+
+# 2- Design experiment / Init plots
+# ---------------------------------
+plt.ion()
```

Comment:
not sure about this, we never use interactive mode. Why is this needed?

Note that the priority is for the examples to render properly in our html docs. If it comes with some tradeoff when running the examples manually, that's something we're OK with

Reply:
This was so that when run the user "sees" it go slow then fast. I can remove it if it is not in the spirit of sklearn examples. Please confirm ?

Reply:
yes please remove

Reply:
ok done